test/suites/dev/test-precompile/test-precompile-multiple-locks.ts

```diff
@@ -0,0 +1,57 @@
+import "@moonbeam-network/api-augment";
+import { beforeAll, describeSuite, expect } from "@moonwall/cli";
+import {
+  GLMR,
+  MIN_GLMR_DELEGATOR,
+  alith,
+  checkBalance,
+  createRawTransfer,
+  generateKeyringPair,
+} from "@moonwall/util";
+import { createProposal } from "../../../helpers/voting.js";
+
+describeSuite({
+  id: "D2580",
+  title: "Precompiles - Locks - Voting and staking locks are not mutually exclusive",
+  foundationMethods: "dev",
+  testCases: ({ context, it }) => {
+    let proposalIndex: number;
+    const randomAccount = generateKeyringPair();
+    const randomAddress = randomAccount.address as `0x${string}`;
+    beforeAll(async function () {
+      await context.createBlock(
+        createRawTransfer(context, randomAddress, MIN_GLMR_DELEGATOR + GLMR)
+      );
+      proposalIndex = await createProposal(context);
+    });
+
+    it({
+      id: "T01",
+      title: "should be able to vote (conviction) and stake",
+      test: async function () {
+        // Vote yes on proposal with 1 GLMR
+        await context.createBlock(
+          context
+            .polkadotJs()
+            .tx.convictionVoting.vote(proposalIndex, { Standard: { Vote: 0, Amount: GLMR } })
+            .signAsync(randomAccount),
+          { allowFailures: false }
+        );
+
+        // Delegate to Alith with MIN_GLMR_DELEGATOR
+        await context.createBlock(
+          context
+            .polkadotJs()
+            .tx.parachainStaking.delegate(alith.address, MIN_GLMR_DELEGATOR, 0, 0)
```

Comment:
`delegate` is [being deprecated](https://github.com/moonbeam-foundation/moonbeam/blob/grw/test-multiple-locks/pallets/parachain-staking/src/lib.rs#L1167), better to use `delegateWithAutoCompound`
```suggestion
            .tx.parachainStaking.delegateWithAutoCompound(
              alith.address,
              MIN_GLMR_DELEGATOR,
              100,
              0,
              0,
              0
            )
```